runtime/src/accounts_db.rs

```diff
@@ -2289,91 +2570,77 @@ impl AccountsDb {
             }
         }
 
-        //TODO: thread this as a ref
-        let mut is_cached = false;
-        let loaded_account = self
-            .get_account_accessor_from_cache_or_storage(slot, pubkey, store_id, offset)
-            .get_loaded_account()
-            .map(|loaded_account| {
-                is_cached = loaded_account.is_cached();
-                (loaded_account.account(), slot)
-            });
+        let (mut account_accessor, slot) = self.retry_to_get_account_accessor(
+            slot,
+            store_id,
+            offset,
+            ancestors,
+            pubkey,
+            max_root,
+            is_root_fixed,
+        )?;
+        let loaded_account = account_accessor.check_and_get_loaded_account();
+        let is_cached = loaded_account.is_cached();
+        let account = loaded_account.take_account();
```

Comment:
So, this is the heart of this pr. And I renamed bunch to make this chunk of code to read best.